### DISCUSSION TITLE
Example on how to use project explorer and git branch

### DISCUSSION THREAD

**Discussion by pabloto** [IBM/vscode-ibmi-projectexplorer#215] [Q&A]
Hi,

I'm using visual studio code and I work in local.

I'd like to improve my developing method, so I'd like to have a default branch (e.g main) with its default setting:

- deploy directory
- library for compile objects, and another for compile files

Then I could have various branch, imagine one for bug fix, and other for new projects, so I'd like to set the deploy directory and library for compile objects and files different from main branch. 

It will be interesting to have a rebuild tools when I make a (merge request/pull request) so every programs/table change in the merged branch will be recompiled starting from main branch.

Do you have an example on how to implement it?

Many thanks.

**Comment by edmundreinhardt** [comment]
Pablo, you have the right idea in mind.
What you want to keep in mind is that the state of all your branches is in git.
You can switch your branches at any time in VS Code using the branches view.
![image](https://github.com/IBM/vscode-ibmi-projectexplorer/assets/11184579/891c3b9c-e3da-4e2b-a2a1-a3cb1901cfdf)

There are 2 ways you can work after that.
You can treat your `Deploy Directory` and `Target Library` as work areas to deploy, compile, debug and test the contents of your branch until it is ready to merge into main.

OR if you want to bounce back and forth between
You can set that deploy location to a dedicated directory for that branch, you can easily change deploy locations using
![image](https://github.com/IBM/vscode-ibmi-projectexplorer/assets/11184579/1ed2aabb-3a99-4149-b09c-95bf19e3bcb3)

or even through though the IFS Browser (note that I have these dragged to the right side window)
![image](https://github.com/IBM/vscode-ibmi-projectexplorer/assets/11184579/05d467e0-a4fe-4ef4-a697-4beb0ff5650e)


If you are taking this approach, you might want to change your target library.  Remember best practice is that this is a library and that you can change it at any time as an environment variable.  Of course the Project Explorer makes this easy
![image](https://github.com/IBM/vscode-ibmi-projectexplorer/assets/11184579/4ffa5adc-c90d-4c9b-aa96-3e13f557385d)
Once it is a variable it is easy to change
![image](https://github.com/IBM/vscode-ibmi-projectexplorer/assets/11184579/f6cb8a38-7ae9-4963-bea2-e7ffbcd96b3a)

This is a great question because it means that you are understanding the value of git!

**Reply by pabloto** [reply]
Hi Edmund, 

It will be interesting if I could create branch from project explorer and in the mean time it will ask to me which is the compile library for the branch and the deploy directory. And this information will be stored into some variables tied to the branch. So if I'm on main branch I compile to prod library and deploy to default directory, while if I checkout to bugfix branch I deploy to bugfix directory and compile to bugfix library.

Is it possible to automate this configuration in the meantime I create bugfix branch?

Thanks 

Bye

**Reply by pabloto** [reply]
Hi @edmundreinhardt ,

What do you think about my proposal?

Thank you

**Reply by SanjulaGanepola** [reply]
Hello @pabloto. Happy to share that branch management is being worked on now and the design ideas are documented in Issue #507. This should fit in well with what you described about having a mapping between branches and libraries to compile into.

**Reply by pabloto** [reply]
Thank you. I'll take a look at it

**Comment by edmundreinhardt** [comment]
Feel free to open an issue to suggest it.  ARCAD does something like that for you and it extends Project Explorer.
In the mean time, you can easily make those  changes directly, it just will take 3 clicks intead of 1.
